assets/blocks/jobs/types.jsx

```diff
@@ -0,0 +1,240 @@
+ */
+const jobTypesAttributesConfig = {
+	showJobTypeFilters: {
+		type: 'boolean',
+		default: true,
+	},
+	includedJobTypes: {
+		type: 'string', // JSON object
+		default: '{}',
+	},
+	allJobTypes: {
+		type: 'string', // JSON array
+		default: '[]',
+	},
+};
+addFilter(
+	'wpjm_block_jobs_attributes_config',
+	'addJobTypesAttributes',
+	( attributesConfig ) => (
+		Object.assign( attributesConfig, jobTypesAttributesConfig )
+	)
+);
+
+/**
+ * Transform a job type attribute value from a JSON-encoded object to a
+ * comma-separated string.
+ *
+ * @param {string} value The JSON-encoded job type attribute value.
+ *
+ * @return {?string} Comma-separated value string for the attribute, or null.
+ */
+function transformJobTypeValue( value ) {
+	let obj;
+	try {
+		obj = JSON.parse( value );
+	} catch ( error ) {
+		// If we cannot parse the value, give up.
+		return null;
+	}
+
+	// Get the keys whose values are truthy.
+	const keysToInclude = _.pickBy( obj, ( val ) => val );
+
+	// Create a comma-separated string.
+	let strValue = Object.keys( keysToInclude ).join( ',' );
+
+	if ( ! strValue ) {
+		strValue = null;
+	}
+
+	return strValue;
+}
+
+/**
+ * Transform the Job Types attributes into shortcode parameters.
+ *
+ * @param {Object} shortcodeParams The incoming shortcode parameters.
+ * @param {Object} attributes      The block attributes.
+ *
+ * @return {Object} The new shortcode parameters.
+ */
+function getShortcodeParameters( shortcodeParams, attributes ) {
+	const jobTypes = transformJobTypeValue( attributes.includedJobTypes );
+	const newShortcodeParams = { ...shortcodeParams };
+
+	if ( attributes.showFilters && attributes.showJobTypeFilters ) {
+		if ( jobTypes ) {
+			newShortcodeParams.selected_job_types = jobTypes;
+		}
+	} else if ( jobTypes ) {
+		newShortcodeParams.job_types = jobTypes;
+	} else {
+		// To make sure the checkboxes are not shown, add all job types to
+		// the list if jobTypes is empty
+		newShortcodeParams.job_types = JSON.parse( attributes.allJobTypes ).join( ',' );
+	}
+
+	return newShortcodeParams;
+}
+addFilter( 'wpjm_block_jobs_shortcode_params', 'getShortcodeParameters', getShortcodeParameters );
+
+/**
+ * UI for filtering job types.
+ */
+class Types extends Component {
+	state = {
+		/**
+	 	 * Initialize object from JSON value in attributes. When this object
+	 	 * gets updates, we will also update the JSON attribute.
+	 	 */
+		includedJobTypes: JSON.parse( this.props.attributes.includedJobTypes ),
+
+		/**
+		 * Keep track of whether we have the API data.
+		 */
+		haveAPIData: false,
+	}
+
+	/**
+	 * Determine whether the API has returned the job types data from the
+	 * server. If we have just received it, update the `allJobTypes` attribute.
+	 *
+	 * @return {boolean} `true` if we have the API data, `false` otherwise.
+	 */
+	haveAPIData() {
+		if ( ! this.state.haveAPIData ) {
+			const { types } = this.props;
+
+			if ( types && ! types.isLoading && 'undefined' !== typeof types.data ) {
```

Comment:
I noticed we never actually dispatch the request/call `types.get()` and so this function always returns `false`.